go/libkb/pgp_key.go

```diff
@@ -150,8 +150,21 @@ func (k *PGPKeyBundle) FullHash() (string, error) {
 
 // StripRevocations returns a copy of the key with revocations removed
 func (k *PGPKeyBundle) StripRevocations() (strippedKey *PGPKeyBundle) {
-	entityCopy := *k.Entity
-	strippedKey = &PGPKeyBundle{Entity: &entityCopy}
+	strippedKey = nil
+	if k.ArmoredPublicKey != "" {
+		// Re-read the key because we want to return a copy, that does
+		// not reference PGPKeyBundle `k` anywhere.
+		strippedKey, _, _ = ReadOneKeyFromString(k.ArmoredPublicKey)
+	}
+
+	if strippedKey == nil {
```

Comment:
I think we will not get that far if it fails to read the key (`StripRevocations` is called from `addKey`)
https://github.com/keybase/client/blob/master/go/libkb/keyfamily.go#L369-L381

But it will not save `ArmoredPublicKey` if there is private key in bundle - https://github.com/keybase/client/blob/master/go/libkb/pgp_key.go#L426-L428